### PR TITLE
Initial app setup

### PR DESCRIPTION
**Pull request by kamwro**
### Major changes regarding this pull requests

1. Added generated project template
2. Installed project dependencies (my aim was to predict all the needed dependecies for the current project module)
3. Added some rules in the files: **tsconfig.json**, **.eslintrc.js**, **.pretttierrc**
4. Added **Swagger UI**
5. Updated **README**
6. Added **Dockerfile**, **docker-compose.yml** and **.dockerignore**

### CODE REVIEW

File: .gitignore
Comment (kamwro) [inline .gitignore:1]: It did not fix the issue at that moment. I needed to untrack the file from the git, which I did in the 2nd PR.

File: .env
Comment (kamwro) [inline .env:1]: I did manage to ignore this file in the 2nd pull request.

File: .gitignore
Comment (kamwro) [inline .gitignore:1]: I might need to include more with the development process, I just wanted it to be more lean.

File: src/app.controller.spec.ts
Comment (kamwro) [inline src/app.controller.spec.ts:1]: Default file.

File: src/app.controller.ts
Comment (kamwro) [inline src/app.controller.ts:1]: Default file.

File: src/app.module.ts
Comment (kamwro) [inline src/app.module.ts:1]: Default file.

File: src/app.service.ts
Comment (kamwro) [inline src/app.service.ts:1]: Default file.

File: src/main.ts
Comment (kamwro) [inline src/main.ts:1]: I only added Swagger. Probably will move Swagger configs to some utils file at some point.

File: test/app.e2e-spec.ts
Comment (kamwro) [inline test/app.e2e-spec.ts:1]: Default file.

File: test/jest-e2e.json
Comment (kamwro) [inline test/jest-e2e.json:1]: Default file.

File: tsconfig.build.json
Comment (kamwro) [inline tsconfig.build.json:1]: Default file.

File: nest-cli.json
Comment (kamwro) [inline nest-cli.json:1]: Default file.

File: tsconfig.json
Comment (kamwro) [inline tsconfig.json:1]: I wanted it as TS strict as possible but will remain open for modifications.

**Review by DominikMarciniszyn — APPROVED** [review]

**Pull request merged**